containers/ecr-viewer/src/app/api/migrate-db/migrations/core/20250509000000_condition_fk.ts

```diff
@@ -0,0 +1,82 @@
+import { Kysely } from "kysely";
+
+import { AnyDb } from "@/app/data/metadataDb/database";
+import { dbNamespace, dbDialect } from "@/app/data/metadataDb/utils/db-config";
+
+/**
+ * Add condition_code column and foreign key constraint to ecr_rr_conditions, backfill condition codes.
+ * @param db - the database connection
+ */
+export async function up(db: Kysely<AnyDb>): Promise<void> {
+  const schema = dbNamespace();
+  const _db = db.withSchema(schema);
+
+  await _db.schema
+    .alterTable("ecr_rr_conditions")
+    .addColumn("condition_code", "varchar(20)")
+    .execute();
+
+  await _db.schema
+    .alterTable("ecr_rr_conditions")
+    .addForeignKeyConstraint(
+      "ecr_rr_conditions_fk_condition_code",
+      ["condition_code"],
+      "condition_reference",
+      ["code"],
+    )
+    .execute();
+
+  // Backfill condition_code for existing records
+  if (dbDialect() === "postgres") {
+    await _db
+      .updateTable("ecr_rr_conditions")
+      .set((eb) => ({
+        condition_code: eb
+          .selectFrom("condition_reference")
+          .select("code")
+          .whereRef(
+            "condition_reference.condition_name",
+            "=",
+            "ecr_rr_conditions.condition",
+          )
+          .limit(1),
+      }))
+      .where("condition_code", "is", null)
+      .execute();
+  } else if (dbDialect() === "sqlserver") {
+    const rows = await _db
+      .selectFrom("ecr_rr_conditions as erc")
+      .innerJoin(
+        "condition_reference as cr",
+        "erc.condition",
+        "cr.condition_name",
+      )
+      .select(["erc.uuid", "cr.code as new_code"])
+      .where("erc.condition_code", "is", null)
+      .execute();
```

Comment:
is there any way we can not have a split on dialect here? does the sqlserver one work for postgres?

Reply:
still open